examples/gno.land/r/sys/users/admin.gno

```diff
@@ -18,40 +16,35 @@ func init() {
 }
 
 // ProposeNewController allows GovDAO to add a whitelisted caller
-func ProposeNewController(addr std.Address) dao.Executor {
-	cb := func() error {
+func ProposeNewController(addr std.Address) dao.ProposalRequest {
+	cb := func(dao.Metadata) error {
 		return addToWhitelist(addr)
 	}
 
-	return bridge.GovDAO().NewGovDAOExecutor(cb)
+	return dao.NewProposalRequestBuilder("Add a new whitelisted caller at sys/users realm").WithCallback(cb).Build()
```

Comment:
I believe the Builder pattern is not suitable in this case. It does not enhance code maintainability and, in this instance, it merely allows for the omission of a description, which I think should be mandatory.

I believe a proposal should always include:
-  a title
-  a description
-  an execution callback

A constructor like `dao.NewProposalRequest(title, desc string, callback ExecutorFunc) ProposalRequest` ensures maximum explicitness, type safety, and straightforwardness.